File: solutions/javascript/difference-of-squares/1/difference-of-squares.js

export class Squares {
    constructor(n) {
        this.n = n
    }

    get sumOfSquares() {
        let sum = 0
        for (let num = 0; num <= this.n; num++) {
            sum += num ** 2
        }               
        return sum    
    }
  
    get squareOfSum() {
        let sum = 0
        for (let num = 0; num <= this.n; num++) {
            sum+=num
        }
        sum  = sum ** 2        
        return sum
    }

    get difference() {
       return this.squareOfSum - this.sumOfSquares
    }
}  

File: solutions/dart/difference-of-squares/1/lib/difference_of_squares.dart
import 'dart:math' as math; 

class DifferenceOfSquares {
   squareOfSum(int n) {
      int sum = 0;
      for (int i = 0; i <= n; i++) sum += i;

      return math.pow(sum, 2);
   }
   
   sumOfSquares(int n) {
      num sum = 0;
      for (int i = 0; i <= n; i++) sum += math.pow(i, 2);
      return sum;     
   }

   differenceOfSquares(int n) => squareOfSum(n) - sumOfSquares(n); 
}